frank-doc-doclet/src/main/java/org/frankframework/frankdoc/properties/PropertyParser.java

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2024 WeAreFrank!
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package org.frankframework.frankdoc.properties;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PropertyParser {
+
+	private static final String START_OR_END_GROUP_TOKEN = "####";
+	private static final String COMMENT_TOKEN = "##";
+	private static final String PROPERTY_TOKEN = "#";
+
+	private static final String[] ALLOWED_FLAGS = new String[]{"Deprecated", "Generated"};
+
+	public List<Group> parse(String fileContent) {
+		ArrayList<Group> groups = new ArrayList<>();
+
+		Group currentGroup = new Group();
+		Property currentProperty = new Property();
+
+		final String[] lines = fileContent.split("\n");
+		for (String line : lines) {
```

Comment:
Zodat je hier 
```suggestion
try (BufferedReader buf = new BufferedReader(reader)) {
		String line = buf.readLine()
		while(line != null) {
				...
				line = buf.readLine();
		}
```
Kan gebruiken. Dan hoef je en niets in geheugen vast te houden, en niet eerst een string te concatenaten en hem vervolgens weer op te splitsen.